From 3f30f472001b80ab6040d5117ae9d78bf4f80957 Mon Sep 17 00:00:00 2001
From: Reto Schneider <reto.schneider@husqvarnagroup.com>
Date: Thu, 4 May 2023 23:46:18 +0200
Subject: [PATCH] [temphack] rtl8xxxu: Print corrupted REG_RCR values

Based on [1] but changed logging level to warn, fixed typo.

[1] https://lore.kernel.org/linux-wireless/d3743b66-23b1-011c-9dcd-c408b1963fca@lwfinger.net/#t
---
 drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c
index 8b4aa32b81bb..c66b8aa1f06e 100644
--- a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c
+++ b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c
@@ -4718,6 +4718,10 @@ rtl8xxxu_bss_info_changed(struct ieee80211_hw *hw, struct ieee80211_vif *vif,
 	u32 val32;
 	u8 val8;
 
+	if (priv->regrcr != rtl8xxxu_read32(priv, REG_RCR)) {
+		pr_warn("REG_RCR corrupted in %s: 0x%x instead of 0x%x\n",
+			__func__, rtl8xxxu_read32(priv, REG_RCR), priv->regrcr);
+	}
 	rarpt = &priv->ra_report;
 
 	if (changed & BSS_CHANGED_ASSOC) {
@@ -6177,6 +6181,10 @@ static void rtl8xxxu_configure_filter(struct ieee80211_hw *hw,
 	struct rtl8xxxu_priv *priv = hw->priv;
 	u32 rcr = priv->regrcr;
 
+	if (priv->regrcr != rtl8xxxu_read32(priv, REG_RCR)) {
+		pr_warn("REG_RCR corrupted in %s: 0x%x instead of 0x%x\n",
+			__func__, rtl8xxxu_read32(priv, REG_RCR), priv->regrcr);
+	}
 	dev_dbg(&priv->udev->dev, "%s: changed_flags %08x, total_flags %08x\n",
 		__func__, changed_flags, *total_flags);
 
-- 
2.39.2

